### ISSUE TITLE
Change Request: Create an `<eslint-editor>` web component

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#15561]
### ESLint version

v8.8.0

### What problem do you want to solve?

Currently the [ESLint demo](https://eslint.org/demo) is custom-built as part of the website.

### What do you think is the correct solution?

We'd like to make a reusable editor web component. It would replace the existing demo, and we could also use it as a built-in playground on rule pages to allow people to play around with different options. It would provide red squigglies and possibly tooltips for errors, but the containing page would be responsible for everything else.

### Participation

- [ ] I am willing to submit a pull request for this change.

### Additional comments

We added this to the roadmap in the 2022-01-27 TSC meeting.

This mostly affects the website, but I put this issue in the main repository initially because I'm guessing a good chunk of the work will be either here or in a new repository.

**Comment by jimmywarting** [comment]
https://github.com/eslint/eslint/issues/15341

**Comment by nzakas** [comment]
@jimmywarting can you provide some context as to why you left that comment?

**Comment by jimmywarting** [comment]
if the point is to make something more web compatible eslint version than i wish there was a possibility to also hook in a own filesystem like the whatwg/fs provided by the browser

**Comment by nzakas** [comment]
Yeah, that’s not really relevant to this effort. What you’re asking for is a change to the ESLint class, which is not possible due to the number of packages that use “fs” directly. What we are talking about here is using the Linter class, which is already runtime agnostic.

**Comment by joshwilsonvu** [comment]
For building web editors, could you add a file exporting the `Linter` class alone to the packages exports public API? Though that class is supposed to be runtime agnostic, it's very difficult to access it in the browser without pulling in Node.js-only code with `import { Linter } from 'eslint'`. 

- [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/website-eslint/src/linter/linter.js) uses `import { Linter } from 'eslint'` for a web demo, but to get that to work they had to build a [custom Rollup plugin](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/website-eslint/rollup-plugin/replace.js) to product an internal package to import elsewhere.
- [the ESLint playground](https://github.com/eslint/eslint.org/blob/main/src/playground/App.js) uses `import { Linter, SourceCodeFixer } from "./node_modules/eslint/lib/linter/"`, which is clearly a workaround for not being able to access `eslint/lib/linter` directly. Seems like a first-class use case.

I'm able to bundle `'./node_modules/eslint/lib/linter/'`, into a file, but there's still a dynamic require, which causes bundle size problems in bundlers that can handle it. TypeScript ESLint does a lot of custom work to handle it. Off-topic but thought I'd mention, please direct me to a better place to say it if necessary.
```js
if (typeof this._rules[ruleId] === "string") {
  this.define(ruleId, require(this._rules[ruleId]));
}
```

**Comment by kecrily** [comment]
I am interested in trying to do this. I noticed that it is labeled as waiting for RFC, and I was wondering what parts of the design this RFC should include?

**Comment by nzakas** [comment]
I think we're looking for an idea of how the component will function:

1. How will configuration be set?
2. What attributes will be defined on the element?
3. What is the progressive enhancement strategy?
4. What editor will be used?
5. How will we support both readonly and writable modes?

Basically, how is it designed to work and be used?
